src/simplewallet/simplewallet.cpp

```diff
@@ -2500,6 +2545,28 @@ bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::stri
 
     switch (transfer_type)
     {
+      case TransferStake:
+        bc_height = get_daemon_blockchain_height(err);
+        if (!err.empty())
+        {
+          fail_msg_writer() << tr("failed to get blockchain height: ") << err;
+          return true;
+        }
+        unlock_block = bc_height + locked_blocks;
+        ptx_vector = m_wallet->create_transactions_2(dsts, fake_outs_count, unlock_block /* unlock_time */, priority, extra, m_trusted_daemon);        
+
+        for (tools::wallet2::pending_tx& tx : ptx_vector)
+        {
+          std::vector<uint8_t>& extra = tx.construction_data.extra;
+          const crypto::secret_key& tx_key = tx.tx_key;
```

Comment:
Any reason to make references here instead of just passing `tx.construction_data.extra` and `tx.tx_key` as arguments directly below?

Reply:
This code has been removed. Transaction signing has been moved to src/cryptonote_core/cryptonote_tx_utils.cpp